src/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -2892,7 +2892,8 @@ public static boolean isObject(Object value)
             return false;
         }
         if (value instanceof ScriptableObject) {
-            return "object".equals(((ScriptableObject)value).getTypeOf());
+            String type = ((ScriptableObject)value).getTypeOf();
+            return "object".equals(type) || "function".equals(type);
```

Comment:
I'm a bit concerned about fixing this problem right here, as opposed to fixing it at the source. "isObject" is in theory a generic thing that's called from lots of places and I'm worried that changing it here might have other effects even if it doesn't break any tests.

Furthermore, I don't really love using "getTypeOf" here since it's kind of expensive versus a pair of instanceof checks.

Reply:
Yes, I now re-read the code and I remember. Thanks for fixing this!